frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts

```diff
@@ -425,42 +433,79 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
                 )
-                if (currentPlayerTime !== null) {
-                    const nextPlayerTime = currentPlayerTime + values.jumpTimeMs
-                    let nextPlayerPosition = getPlayerPositionFromPlayerTime(
-                        nextPlayerTime,
-                        values.sessionPlayerData.metadata.segments
-                    )
-                    if (!nextPlayerPosition) {
-                        // At the end of the recording. Pause the player and reset the playerPosition
-                        actions.setPause()
-                        nextPlayerPosition = values.sessionPlayerData.metadata.segments[0].startPlayerPosition
-                    }
-                    actions.seek(nextPlayerPosition)
+                if (!nextPlayerPosition) {
+                    // At the end of the recording. Pause the player and set to the end of the recording
+                    actions.setPause()
+                    nextPlayerPosition = values.sessionPlayerData.metadata.segments.slice(-1)[0].endPlayerPosition
                 }
+                actions.seek(nextPlayerPosition)
             }
         },
         seekBackward: () => {
-            if (values.currentPlayerPosition) {
-                const currentPlayerTime = getPlayerTimeFromPlayerPosition(
-                    values.currentPlayerPosition,
+            if (!values.currentPlayerPosition) {
+                return
+            }
+
+            const currentPlayerTime = getPlayerTimeFromPlayerPosition(
+                values.currentPlayerPosition,
+                values.sessionPlayerData.metadata.segments
+            )
+            if (currentPlayerTime !== null) {
+                const nextPlayerTime = Math.max(currentPlayerTime - values.jumpTimeMs, 0)
+                const nextPlayerPosition = getPlayerPositionFromPlayerTime(
+                    nextPlayerTime,
                     values.sessionPlayerData.metadata.segments
                 )
-                if (currentPlayerTime !== null) {
-                    const nextPlayerTime = Math.max(currentPlayerTime - values.jumpTimeMs, 0)
-                    const nextPlayerPosition = getPlayerPositionFromPlayerTime(
-                        nextPlayerTime,
-                        values.sessionPlayerData.metadata.segments
-                    )
-                    actions.seek(nextPlayerPosition)
-                }
+
+                actions.seek(nextPlayerPosition)
+            }
+        },
+
+        seekBackwardOneFrame: () => {
+            if (!values.currentPlayerPosition || !values.currentPlayerTime) {
+                return
             }
+            actions.setPause()
+
+            const nextPlayerTime = Math.max(values.currentPlayerTime - ONE_FRAME_MS)
+            const nextPlayerPosition = getPlayerPositionFromPlayerTime(
+                nextPlayerTime,
+                values.sessionPlayerData.metadata.segments
+            )
+            actions.seek(nextPlayerPosition)
```

Comment:
Nit: It seems like this logic is used in four different places now. Might be worth refactoring this into a util or separate action.

Reply:
Yeah thats a good point. I felt like they were all very similar but not at that same time 😅 I'll have another pass and see if it can be simplified, otherwise will leave that to a future improvment